### DISCUSSION TITLE
Translation - Header title in admin modules

### DISCUSSION THREAD

**Discussion by HoenesDK** [oqtane/oqtane.framework#3094] [Q&A] [answered]
How do i get the header title in the Admin modules translated.
I have added the translations to SharedResources.da.resx, but header title is not translated.
![TranslationProblem](https://github.com/oqtane/oqtane.framework/assets/51191152/3a362bf9-fe55-40fa-9eda-be780c44845b)

**Comment by sbwalker** [comment]
@HonesDK #3117 should fix the issue you raised above... it appears that although the translations from SharedResources.resx were being used properly in the Admin Dashboard, the logic in the Module Title control was missing.

**Comment by sbwalker** [comment]
@leigh-pointer were you aware of this issue? I am just curious if I am missing something as I am surprised this has not been reported previously.

**Reply by leigh-pointer** [reply]
@sbwalker I wasnt aware.  Seems strange how that slipped through!  Are we sure that it has always been that way ?

**Reply by alikoli** [reply]
There is still an issue with section headings. For example Appearance is not taken from the resource, while all the properties (logo, etc) are translated correctly.

**Reply by sbwalker** [reply]
#3119 fixes the issue with the Section component. Note that keys need to be named ResourceKey.Heading

**Reply by alikoli** [reply]
So cool, now most translation issues are gone. With Appearance there is still an issue but in different place. I double checked if I have all current keys translated and went through Admin pages. Here are some issues I've found:

![image](https://github.com/oqtane/oqtane.framework/assets/8749335/e2ef753f-c917-478e-bbf9-9d60e777370a)

![image](https://github.com/oqtane/oqtane.framework/assets/8749335/82cdb30f-88ec-4c50-afdb-de3a163e7c2d)

![image](https://github.com/oqtane/oqtane.framework/assets/8749335/0383a86c-ebcf-4ac2-94cc-5e80cef0fcdb)
![image](https://github.com/oqtane/oqtane.framework/assets/8749335/20eb0982-9d0e-4757-9896-d5660c2b2c1d)
![image](https://github.com/oqtane/oqtane.framework/assets/8749335/099de9af-e714-4519-aa9d-765d354c1ab4)
![image](https://github.com/oqtane/oqtane.framework/assets/8749335/e4c893a4-afcf-4fd7-ac65-a4544e5030d2)
![image](https://github.com/oqtane/oqtane.framework/assets/8749335/1aef1b0e-a3c6-4d96-8d87-872c16290e91)
![image](https://github.com/oqtane/oqtane.framework/assets/8749335/76aece75-83d2-467f-bf88-be10e45bc4fb)
![image](https://github.com/oqtane/oqtane.framework/assets/8749335/7b2a185e-8633-434b-89d0-a8f4aa940d70)
![image](https://github.com/oqtane/oqtane.framework/assets/8749335/08403be6-9a07-459a-935a-9d10d29abbe5)
![image](https://github.com/oqtane/oqtane.framework/assets/8749335/d0fcbf5a-c362-47ed-8778-2f284d409c5e)

**Reply by alikoli** [reply]
And these are when I use edit current page:
![image](https://github.com/oqtane/oqtane.framework/assets/8749335/1c9f3b9e-97d3-4150-a248-8f82c696eab6)
![image](https://github.com/oqtane/oqtane.framework/assets/8749335/92361333-11e0-4fcd-a2f0-524562885555)
![image](https://github.com/oqtane/oqtane.framework/assets/8749335/91e731d2-dc68-43bd-8c44-7cba6e68fefb)

**Comment by sbwalker** [comment, marked as answer]
@HonesDK thank you very much for identifying where localization issues exist. I do not believe that anyone has done a thorough localization review in recent releases, so it is not surprising that there are some missing keys, etc... The good news is that these are relatively simple issues to resolve - in fact @leigh-pointer has already addressed most of them in the past 12 hours - so they will be included in the upcoming 4.0.2 release.

**Reply by alikoli** [reply]
Let me know when they are fixed in a dev branch so I will verify all of them once again.

**Reply by sbwalker** [reply]
@alikoli they are now fixed in the Dev branch

**Reply by alikoli** [reply]
I confirm that some of the problems are solved. There is NullReferenceException on the Line:
public override string Title => Localizer["FolderManagement.Title"];
in oqtane.framework\Oqtane.Client\Modules\Admin\Files\Edit.razor
When I click EditFolder or AddFolder button. This problem is a new one, because I checked these buttons yesterday.

It looks like there are still some issues:
![image](https://github.com/oqtane/oqtane.framework/assets/8749335/a5fa2072-546e-4a2a-aee1-e007bc68ecd7)
![image](https://github.com/oqtane/oqtane.framework/assets/8749335/9da80af6-0a8c-47c4-b97e-8c2b7001590b)
![image](https://github.com/oqtane/oqtane.framework/assets/8749335/4415b4e6-e9e6-43dd-9793-f2697674b24f)
![image](https://github.com/oqtane/oqtane.framework/assets/8749335/d2a1748d-47e4-4347-9e8a-ed8921ff0719)
![image](https://github.com/oqtane/oqtane.framework/assets/8749335/b0bb3c45-f5f4-4690-bbb1-4473d2771dbe)
![image](https://github.com/oqtane/oqtane.framework/assets/8749335/4773b158-727d-4bed-8941-52ed88a37da2)
![image](https://github.com/oqtane/oqtane.framework/assets/8749335/0952981d-2254-453a-95e4-30f3cf3b2fd2)
![image](https://github.com/oqtane/oqtane.framework/assets/8749335/129a67a5-3df7-47ee-8372-3792389272d5)

**Comment by sbwalker** [comment]
@leigh-pointer you cannot use the Localizer in this scenario as the Localizer service has not yet been instantiated. This throws a null reference. We will need to look through your recent PR to revert any of these changes.

```
    public override string Title => Localizer["FolderManagement.Title"];
```

**Reply by leigh-pointer** [reply]
@sbwalker there was only one! Fixed with new PR

**Reply by leigh-pointer** [reply]
@sbwalker I can fix this up for all (currently 8) places where the Module Title is overriden with the base.SetModuleTitle

> base.SetModuleTitle(Localizer["FolderManagement.Title"]);

**Reply by sbwalker** [reply]
@leigh-pointer yes, please go ahead. I don't think you need to include the "base." as the method is inherited.

**Reply by alikoli** [reply]
I fixed some more issues with translation, could you merge my PR?

**Comment by alikoli** [comment]
Now this error is gone.
Edit folder:
![image](https://github.com/oqtane/oqtane.framework/assets/8749335/45edc398-5208-4100-b97c-45e43d34fe8a)
Add folder:
![image](https://github.com/oqtane/oqtane.framework/assets/8749335/e15bf415-996b-48cf-9213-478c64f1ab5c)
This looks strange when part of the string is translated and the rest remain in english:
![image](https://github.com/oqtane/oqtane.framework/assets/8749335/07e50bfd-3d82-4d74-9233-5c6eceedbbe3)
